cdap-common/src/main/java/co/cask/cdap/internal/app/store/RunRecordMeta.java

```diff
@@ -50,15 +51,26 @@
   @Nullable
   private final byte[] sourceId;
 
+  @SerializedName("artifactId")
+  @Nullable
+  private final ArtifactId artifactId;
+
+  @SerializedName("principal")
+  @Nullable
+  private final String principal;
+
   private RunRecordMeta(ProgramRunId programRunId, long startTs, @Nullable Long runTs, @Nullable Long stopTs,
                         ProgramRunStatus status, @Nullable Map<String, String> properties,
                         @Nullable Map<String, String> systemArgs, @Nullable String twillRunId,
-                        ProgramRunCluster cluster, byte[] sourceId) {
+                        ProgramRunCluster cluster, byte[] sourceId,
+                        @Nullable ArtifactId artifactId, @Nullable String principal) {
```

Comment:
artifactId for new run records should never be null right? It can only be null from the getter for old run records?